src/main.ts

```diff
@@ -164,12 +164,12 @@ export function activate(context: vscode.ExtensionContext): ReturnType<typeof ge
             extension.logger.addLogMessage(`onDidSaveTextDocument triggered: ${e.uri.toString(true)}`)
             extension.manager.updateCachedContent(e)
             extension.linter.lintRootFileIfEnabled()
-            extension.structureViewer.update()
+            void extension.structureViewer.update()
```

Comment:
How about moving the line to src/providers/structure.ts?

Reply:
It is possible to register this line in a call to `onDidSaveTextDocument` in `src/providers/structure.ts` but then it does not really make sense to log `onDidSaveTextDocument triggered`, does it?

Reply:
Actually this is not so easy to move the line to an other call to `onDidSaveTextDocument` because `structureViewer.update` relies on the value of the `CachedContent`. We must make sure `extension.manager.updateCachedContent(e)` has finished first.

Reply:
We have an implicit dependency here. It would increase the maintenance burden. One of the ways to make the dependency explicit is using `EventBus`. `updateCachedContent` fires an event, and the structure component calls `update` on the event.

At least we had better write a comment on the dependency.

Reply:
I agree that we had better use `EventBus` to make dependency explicit.
I am afraid there are a lot of such dependencies in LW. This should be handled in a separate PR.